pkg/storage/raft.go

```diff
@@ -99,14 +100,42 @@ func (r *raftLogger) Fatalf(format string, v ...interface{}) {
 func (r *raftLogger) Panic(v ...interface{}) {
 	s := fmt.Sprint(v...)
 	log.ErrorfDepth(r.ctx, 1, s)
+	if is14231Error(s) {
+		log.Shout(r.ctx, log.Severity_ERROR, is14231HelpMessage)
+	}
 	panic(s)
 }
 
 func (r *raftLogger) Panicf(format string, v ...interface{}) {
-	log.ErrorfDepth(r.ctx, 1, format, v...)
+	s := fmt.Sprintf(format, v...)
+	log.ErrorfDepth(r.ctx, 1, s)
+	if is14231Error(s) {
+		log.Shout(r.ctx, log.Severity_ERROR, is14231HelpMessage)
+	}
 	panic(fmt.Sprintf(format, v...))
 }
 
+// Returns whether the provided error message matches the error observed in
+// issue #14231. Such errors are likely to be the result of wiping a store's
+// data directory and bringing it back online without join flags but with the
+// same address, node ID, and store ID, and join flags. While we should prevent
+// these errors in our next release, for now the best we can do is help users
+// understand them.
+//
+// TODO(#14231): Remove this method once the issue is fixed (hopefully in 1.2).
+func is14231Error(s string) bool {
+	return strings.Contains(s, "is out of range [lastIndex(") &&
+		strings.Contains(s, "Was the raft log corrupted, truncated, or lost?")
+}
+
+// TODO(#14231): Remove this once the issue is fixed (hopefully in 1.2).
+const is14231HelpMessage = "Server crashing due to missing data. Was the server restarted with a " +
```

Comment:
log.Shout supports multi-line messages. You can wrap this one.

Reply:
Done.